ledger-core/virtual/integration/deduplication/constructor_test.go

```diff
@@ -24,118 +24,147 @@ import (
-				if test.sendMessageFn != nil {
-					test.sendMessageFn()
-				}
-				return false
-			})
-
-			{
-				server.SendPayload(ctx, &pl)
-			}
-
-			testutils.WaitSignalsTimed(t, 10*time.Second, synchronizeExecution.Wait())
-
-			{
-				server.SendPayload(ctx, &pl)
-			}
-
-			synchronizeExecution.WakeUp()
-
-			testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
-			testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
-
-			{
-				assert.Equal(t, test.resultCount, typedChecker.VCallResult.Count())
-			}
-
-			mc.Finish()
 
+	{
+		requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+		requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+		executionMock := runnerMock.AddExecutionMock(outgoing.String())
+		executionMock.AddStart(executionFn, &execution.Update{
+			Type:   execution.Done,
+			Result: requestResult,
 		})
 	}
+
+	pl := payload.VCallRequest{
+		CallType:       payload.CTConstructor,
+		CallFlags:      payload.BuildCallFlags(isolation.Interference, isolation.State),
+		Callee:         class,
+		CallSiteMethod: "New",
+		CallOutgoing:   outgoing,
+	}
+
+	{
+		// send first call request
+		server.SendPayload(ctx, &pl)
+	}
+
+	// await first SMExecute go to step execute (in this point machine is still not publish result to table in SMObject)
+	testutils.WaitSignalsTimed(t, 10*time.Second, synchronizeExecution.Wait())
+
+	{
+		// send second call request
+		server.SendPayload(ctx, &pl)
+	}
+
+	// second SMExecute should stop in deduplication algorithm and she should not send result because she started during execution first machine
+	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitStopOf(&execute.SMExecute{}, 1))
+
+	// wakeup first SMExecute
+	synchronizeExecution.WakeUp()
+
+	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitStopOf(&execute.SMExecute{}, 1))
+	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+
+	{
+		assert.Equal(t, 1, typedChecker.VCallResult.Count())
+	}
+
+	mc.Finish()
+}
+
+func TestConstructor_SamePulse_AfterExecution(t *testing.T) {
+	t.Log("C5005")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, nil)
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	var (
+		isolation = contract.ConstructorIsolation()
+		outgoing  = server.BuildRandomOutgoingWithPulse()
+		class     = gen.UniqueGlobalRef()
+	)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+		return false
+	})
+
+	{
+		requestResult := requestresult.New([]byte("123"), gen.UniqueGlobalRef())
+		requestResult.SetActivate(gen.UniqueGlobalRef(), class, []byte("234"))
+
+		executionMock := runnerMock.AddExecutionMock(outgoing.String())
+		executionMock.AddStart(func(ctx execution.Context) {
```

Comment:
replace with nil, pls, if you won't check anything

Reply:
Done